brave/src/main/java/brave/RealScopedSpan.java

```diff
@@ -0,0 +1,46 @@
+package brave;
+
+import brave.internal.recorder.Recorder;
+import brave.propagation.TraceContext;
+import com.google.auto.value.AutoValue;
+
+/** This wraps the public api and guards access to a mutable span. */
+@AutoValue
+abstract class RealScopedSpan extends ScopedSpan {
```

Comment:
note.. you may realize this doesn't actually scope yet :) I raised early just to show the shape of the api